modules/site-health/audit-autoloaded-options/load.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * Module Name: Audit Autoloaded Options
+ * Description: Adds Autoloaded options checker in Site Health checks.
+ * Experimental: No
+ *
+ * @package performance-lab
+ * @since 1.0.0
+ */
+
+/**
+ * Adds test to site health.
+ *
+ * @since 1.0.0
+ *
+ * @param array $tests Site Health Tests.
+ * @return array
+ */
+function perflab_aao_add_autoloaded_options_test( $tests ) {
+	$tests['direct']['autoloaded_options'] = array(
+		'label' => esc_html__( 'Autoloaded options', 'performance-lab' ),
+		'test'  => 'perflab_aao_autoloaded_options_test',
+	);
+	return $tests;
+}
+add_filter( 'site_status_tests', 'perflab_aao_add_autoloaded_options_test' );
+
+/**
+ * Callback for autoloaded_options test.
+ *
+ * @since 1.0.0
+ *
+ * @return array
+ */
+function perflab_aao_autoloaded_options_test() {
+
+	$autoloaded_options_size  = perflab_aao_autoloaded_options_size();
+	$autoloaded_options_count = count( wp_load_alloptions() );
+
+	$result = array(
+		'label'       => esc_html__( 'Autoloaded options', 'performance-lab' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => esc_html__( 'Performance', 'performance-lab' ),
+			'color' => 'blue',
+		),
+		'description' => sprintf(
+		/* translators: 1: Number of autoloaded options. 2.Autoloaded options size. */
+			'<p>' . esc_html__( 'The amount of %1$s autoloaded options (size: %2$s) in options table is acceptable.', 'performance-lab' ) . '</p>',
+			$autoloaded_options_count,
+			size_format( $autoloaded_options_size )
+		),
+		'actions'     => '',
+		'test'        => 'autoloaded_options',
+	);
+
+	/**
+	 * Hostings can modify its limits depending on their own requirements.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int $limit Autoloaded options threshold size. Default 800000.
+	 */
+	$limit = apply_filters( 'perflab_aao_autoloaded_options_limit_size_in_bytes', 800000 );
+
+	if ( $autoloaded_options_size < $limit ) {
+		return $result;
+	}
+
+	$result['status']         = 'critical';
+	$result['badge']['color'] = 'red';
+	$result['description']    = sprintf(
+	/* translators: 1: Number of autoloaded options. 2.Autoloaded options size. */
+		'<p>' . esc_html__( 'Your website uses %1$s autoloaded options (size: %2$s). Try to reduce the number of autoloaded options or performance will be affected.', 'performance-lab' ) . '</p>',
```

Comment:
How can the user reduce the number of autoloaded options? The average end user will have no idea what that even means. We have introduced extensive documentation even for high-level technical topics like [what PHP is and how to update it](https://wordpress.org/support/update-php/#why-php-matters-to-you), and I'd argue the topic here goes way beyond that.

Also, there is no way for the user right now to see the autoloaded options, so even if they had an idea about what it means, how can they reason about it or fix it? In the "Audit Enqueued Assets" module for example, people can go to the frontend and see which CSS and JS assets are being enqueued and then based on that get an idea which plugins or themes are responsible. But that is not possible here.

Reply:
Hello @felixarntz those are valid concerns, my initial idea was to flag this problem so hostings could link to actionable guides on how to solve it ( you can check the original discussion on its [issue](https://github.com/WordPress/performance/issues/92))

Most, if not all, hosting providers have specific articles pointing to this problem, see [Kinsta](https://kinsta.com/knowledgebase/wp-options-autoloaded-data/) one.

It happened to me in the past that some clients were having death screens that were caused by this, opening a ticket to the hosting, it ended up sending me a link to its guide on how to solve it. These warnings would allow hosting to tweak its limits and point the client in the right direction. (that's why threshold + actionable text is filterable).

Regarding default behavior, @tillkruss mentioned in the issue that this would be a nice place [https://wordpress.org/support/article/optimization/](https://wordpress.org/support/article/optimization/) though it would need a new topic to cover the issue.

As a next iteration, indeed, we should expand functionality and give more information or even offer some actionable solution, see again point 2 in the issue.
> Functional module, we can go a step further and keep track of used/unused options, and then give the ability to users to clean up the table from the site health screen itself.

Reply:
Let's continue this discussion on the issue as you've already started doing. The main thing if we don't add the guidance here is that there should be another issue to do it.

Will resolve this comment in favor of conversation on the issue (where there's also the vote going on).